### ISSUE TITLE
Rationale behind multiplying by sinc2 grid in `project_3d_to_2d`

### ISSUE THREAD

**Issue by mgiammar** [teamtomo/torch-fourier-slice#40]
@alisterburt I remember you having some rationale (which I can't seem to remember) behind the sinc-squared grid multiplication during slice extraction:

https://github.com/teamtomo/torch-fourier-slice/blob/b53a76e0ae56e8086a36d929b6bd3ca4aeb149f1/src/torch_fourier_slice/project.py#L57-L65

I'm trying to re-update my mental model of how `torch-fourier-slice` is operating since there's some downstream computations whose numerical stability depends somewhat on having accurate projections. Would you mind re-explaining the rationale?

**Comment by alisterburt** [comment]
could you compare to a straight projection of the map by summing along the z dimension instead of no sinc? That's a stronger control

Would also be good to see how pre division by sinc2 looks compared to pre multiplication

**Comment by alisterburt** [comment]
Might also be worth comparing directly to projection code in relion/warp

**Comment by alisterburt** [comment]
Ah I just meant summing the simulated map (input for this library) along the depth dim e.g. torch.sum(volume, dim=-3)

**Comment by alisterburt** [comment]
@michael-0brien great to see you here! Happy holidays, just wanted to say I'm super impressed by everything you're doing with cryojax :-)

I think it's pretty clear that the multiplication by sinc^2 here is a bug and dividing by sinc^2 in real space prior to dft calculation is closer to analytically correct

Useful note re: invariants - thank you - I guess ideally the signal scale is the same as you would get from a true projection and the spherical mask is to avoid artifacts from lack of periodicity in fourier space...

FWIW I've always seen this `project_3d_to_2d()` as a useful convenience layer but you often have to drop down to `extract_central_slices_rfft_3d()` for perf/specific control depending on the specific application, I don't think there is one right answer for how weighting should be handled 🙂

**Comment by michael-0brien** [comment]
> [@michael-0brien](https://github.com/michael-0brien) great to see you here! Happy holidays, just wanted to say I'm super impressed by everything you're doing with cryojax :-)

Much appreciated! Same goes for teamtomo, we rely on a few of these packages. `mmdf` was a lifesaver for us.

> I think it's pretty clear that the multiplication by sinc^2 here is a bug and dividing by sinc^2 in real space prior to dft calculation is closer to analytically correct

Yes---just note also the difference with cisTEM in computing $sinc(|r|)$ vs $sinc(r_x) * sinc(r_y) * sinc(r_z)$. If I'm not mistaken I think this difference is somewhat significant. This comes from the 3D fourier transform of the unit box.

> Useful note re: invariants - thank you - I guess ideally the signal scale is the same as you would get from a true projection and the spherical mask is to avoid artifacts from lack of periodicity in fourier space...

Yes, makes sense. Whether or not to do this seems quite dependent on the analysis pipeline!

> FWIW I've always seen this `project_3d_to_2d()` as a useful convenience layer but you often have to drop down to `extract_central_slices_rfft_3d()` for perf/specific control depending on the specific application, I don't think there is one right answer for how weighting should be handled 🙂

Agreed! We take a similar philosophy. It can be tricky to tie users to a fixed pattern for how diverse analysis pipelines can be. We also implement cubic interpolation which actually does a decent job handling the scaling issue, but this comes with its own significant downsides...

**Comment by mgiammar** [comment]
@michael-0brien I recently took a peek at the cryoJAX work and it's very exciting. Glad there are even more people out there interested in developing good open-source scientific software out there.

I have not had the brain-space recently to properly think about what prefilters are "correct" to apply, only that some projection results didn't match exactly what I expected and needed to know why. With the separability of  N-dim FFTs, I'd expect the the `sinc(r_x) * sinc(r_y) * sinc(r_z)` to be proper, but that _is not_ what the plots I've shared above were doing. I'll test this out when I get a chance.

**Comment by michael-0brien** [comment]
> [@michael-0brien](https://github.com/michael-0brien) I recently took a peek at the cryoJAX work and it's very exciting. Glad there are even more people out there interested in developing good open-source scientific software out there.

Thanks so much, back at you!

> I have not had the brain-space recently to properly think about what prefilters are "correct" to apply, only that some projection results didn't match exactly what I expected and needed to know why. With the separability of N-dim FFTs, I'd expect the the `sinc(r_x) * sinc(r_y) * sinc(r_z)` to be proper, but that _is not_ what the plots I've shared above were doing. I'll test this out when I get a chance.

Makes sense! Yes the separability is a good way to see it. Thanks so much for testing this out, I'm super interested to see what kinds of differences it makes.
